Big Update to Bay (OK TO MERGE)

PsyCommando opened the pull request:
***Don't merge***
_Currently, power nets don't work properly, and the char creation menu's layout is a bit de-faced and broken. But I'm gonna have everything fixed._

This PR brings the codebase more up to date with bay, and adds a lot of the missing features, items, and improvements that were made on bay while merging with our code, and making the code base easier to merge with future bay updates, hopefully.

* Adds fluid system.
* Adds, char setup system
* Removes power use recalc every ticks, and use the new method by area to compute power usage.
* Adds missing reagents.
* Adds updated sprites + moved everything over to the new mob sprite file layout.
* Added augments
* Updated surgery to latest system.
* Added psionics.
* Updated the overmap + ship and shuttle system. (Didn't touch buildable shuttles, so that might need some more testing)
* Used improved mob code.
* Added pre-TG circuits module (currently de-activated, because there's some choices to be made there..)
* Fixed john breaking my prs :P
* Renamed subsystem files back to the bay format, because merging them otherwise is an enormous pain in the ass, and they're some of the more likely things to be updated to bay in the future.
* (More stuff to be added to the desc later, but I don't have all day to write this, and if I keep putting back writing the PR this is never gonna get done..)

And yes, I went through every single files one by one comparing them to baycode and merged everything at a snail's pace. There might be some bugs though, so testing is appreciated.

PsyCommando committed:
Fix bad parameters being passed to damage functions

* Fix bad parameters being passed to take_damage and apply_damage
* Add missing damageable flag
* Misc little tweaks

PsyCommando committed:
Add missing DAM_SHATTER damage type

PsyCommando committed:
WIP porting of bay's updates

PsyCommando committed:
wip ported most things we needed from bay. Now just fixing compilation

PsyCommando committed:
More progress

PsyCommando committed:
...

PsyCommando committed:
Most runtimes fixed

* Main issue is that carbon mobs just slowly die for some reasons.

PsyCommando committed:
Merge remote-tracking branch 'upstream/dev' into DamageFixes

PsyCommando committed:
Icon fixes, runtime fixes, etc

PsyCommando committed:
Merge remote-tracking branch 'upstream/dev' into DamageFixes

PsyCommando committed:
Fixed some runtimes + extra bay stuff

- Powernets still don't process correctly, but it seems to be some sort of race condition, because the power gen count seems to be cleared before its processed..
- Character menu is broken. It seems there's 2 character menus running in parallel and only one of them seems to save..
* Merged in the latest dev branch from bay.

PsyCommando committed:
merged changes from my laptop

PsyCommando committed:
added back the paint gun since some things seems to have changed

PsyCommando committed:
removed leftover merge symbols..

PsyCommando committed:
Fixed powernets + fixed a lot of issues

* Still seems to be some mob damage issues after some of the changes from bay were merged.
* Added tiles for new floor types
* Added buildable floorlights and navlight.
* Tweaked the map a bunch to fit in some of the new floor types, decals, lights, etc..
* Added a small disposal for the government floor only, since it could get bad quickly without trash disposal.
* Fixed tons of missing icons, still a bunch left.
* Made the machine loop sound I added long ago for the recycler less annoying.

PsyCommando committed:
wip character creation menu fix

ingles98 commented on code/_helpers/global_access.dm:
```diff
@@ -783,6 +731,8 @@
 			return global.priv_syndicate_access;
 		if("prometheans")
 			return global.prometheans;
+		if("prometheans")
```
Was this generated ?

PsyCommando replied:
Globals will have to be re-generated later

PsyCommando committed:
merged bay language changes

PsyCommando committed:
Merge branch 'DamageFixes' of https://github.com/PsyCommando/Persistent-Bay into DamageFixes

PsyCommando committed:
pushed more updates from bay + wip upgrade of the crewreport

PsyCommando committed:
merged more bay update stuff + fixed compilation

- Crew records might be slightly broken

PsyCommando committed:
possible wireless fix

PsyCommando committed:
wip fixing icon issues, misc, and player creation menu

PsyCommando committed:
progress on fixing char spawning

* gotta finish /datum/preferences/proc/create_mannequin()

PsyCommando committed:
Fixed characters spwaning

- Character preview is still not working properly.
* Made autosave and election subsystem run only on RUNLEVEL_GAME since its basically just wasting CPU for nothing during init.
* Made the character save paths macros a bit more coherent.
* Made sure ambiences, voting sounds, and lobby music all use the proper global sound channels from the repository.
* Fixed despawning in the cryopods not working.
* Made the plated catwalk floors save properly.
* Re-organized character save/load procs so they're in the character_setup subsystem for possible future expension.
* Improved saving/loading for human mobs, and defined saved variables.
* Moved spawn cutscene to map specific code for future customization.
* Made character creation option hidden until initialization is complete to prevent runtimes and other issues.
* Merged manually missing clothing icons from bay.
* Tweaked default faction outfits for the nexus.

ingles98 commented:
Mega hypes!

PsyCommando committed:
Fixed character creation

* fixed preview icon displaying empty sprites
* Fixed mob spawning/saving naked or without
* Fixed the newscaster filling the chat window completely on the entire station with news update messages.
* Fixed invisible underwears

NataKilar commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -765,8 +924,8 @@
-	color = "#cccccc"
-	metabolism = 0.1
+	color = COLOR_GRAY80
+	metabolism = 0.05 // So that low dosages have a chance to build up in the body.
```
I assume this change was accidentally included from a Bay cherrypick. Metabolism (and some functioning) of N2O was changed in https://github.com/Persistent-SS13/Persistent-Bay/pull/949/. I highly recommend leaving it at 0.1 or we're going to end up with people sleeping until the heat death of the universe again.

PsyCommando replied:
Done.

NataKilar commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Other.dm:
```diff
@@ -330,60 +405,57 @@
 	var/warning_prob = 10
 	var/dosage = M.chem_doses[type]
-	if(dosage >= 3 * REAGENT_GAS_EXCHANGE_FACTOR)
+	if(dosage >= 3)
```
This was previously changed at numerous players' requests to prevent mass asphyxiation in closed off rooms. The multiplication by REAGENT_GAS_EXCHANGE_FACTOR was ideal because it would automatically adjust with other balancing of gaseous reagents entering the blood.

PsyCommando replied:
It seems pretty arbitrary though. It would be nice if you guys would leave comments to explain obscure fixes like that XD

NataKilar commented on code/modules/reagents/Chemistry-Sublimator.dm:
```diff
@@ -0,0 +1,162 @@
+/proc/create_gas_data_for_reagent(var/datum/reagent/reagent)
```
Obviously gaseous reagents are in a very precarious spot with SoM anyway but it should be noted that if the sublimator is to be made obtainable, than create_data_gas_for_reagent should probably be left in xgm_gas_data.dm as it was before in order to prevent needing to save what reagent gases have been created. The impact on start-up time was negligible.

PsyCommando replied:
Reagent gases are still generated on startup, that proc won't do anything if the reagent gas exists already. And I'd rather not start butchering code when I'm close to merging this.

PsyCommando committed:
Fixed a bunch of things

* fixed really long atmos_init now the game will skip deleted stuff.
* removed duplicate entry for cloth material
* made some stuff buildable again(showers and etc)
- air alarms lose their connnection to vents on load.
- still trying to fix records and ids init
- stairs poke holes through levels right now... so gotta change that behavior

PsyCommando committed:
Fixed records, and id card info

There was just a timing issue
* Increased a tiny bit the lighting range on lights.
* Set the whitelisted flags on lots of species so they're not freely available.
* Made sure characters spawn with their laces.
- for some reasons the uniform selection won't update correctly with the origin selected...
- If a human spawn with something in their hands it'll be invisible for some reasons until dropped..

PsyCommando committed:
prototype fix for pref saving + some reagents fixing for reagent gases

PsyCommando committed:
custom radio fix attempt

PsyCommando committed:
Many fixes, now going to sleep

PsyCommando committed:
Fixed phorosian culture + fixed some missing chair icons + fixed missing tanks icon in suit storage

* Fixed stairs moving on save.
* Fixed air vents not saving properly
* Fixed LTC computers not saving properly.
* Made several things save better. Also added loads of saved vars via code.
* MAde a lot of machines use the new method of populating parts using the circuit board's part list itself instead of having to duplicate it in the machine's constructor.

PsyCommando committed:
Save improvements + insane power usage with some machines fixed

* Made many more machines use the automated part install system.
* Made lots of reagents containers not respawn their content on instanciation
* Fixed the Process proc on HE pipes not rebuilding their pipe net and causing them to turn into a buggy mess on save load.

PsyCommando committed:
More saving stuff

PsyCommando committed:
Nata fixes

PsyCommando committed:
Merge remote-tracking branch 'upstream/dev' into DamageFixes

PsyCommando committed:
Merge fiix + fixed spawning

PsyCommando commented:
So, prefs still don't save but besides that things are mostly working. Except some classic atmos bs..

PsyCommando commented:
Since people are waiting on this, lets just merge it, and we'll handle the fixes for preferences later, I guess? Its probably something really dumb I'm missing, since the save path string end up being null for no reasons..

PsyCommando committed:
Removed a lot of garbage testing output + remove redundant stuff in light tiles

PsyCommando committed:
MapMergePass

PsyCommando committed:
fixed intercom icon being off on a wall

PsyCommando committed:
set the +x flag on run-test.sh

PsyCommando committed:
A sacrifice for satisfying the travis gods

PsyCommando committed:
More travis fixes (indentation + duplicated include)

PsyCommando committed:
Fix indentation again

Andrew-Fall committed:
Merge branch 'dev' into DamageFixes

Andrew-Fall committed:
Update records.dm

Andrew-Fall committed:
Merge branch 'dev' into DamageFixes

Andrew-Fall committed:
Merge branch 'dev' into DamageFixes
